### DISCUSSION TITLE
Should we require that Parameters and Descriptors are given a name?

### DISCUSSION THREAD

**Discussion by henrikjacobsenfys** [easyscience/core#46] [Ideas]
Currently, Parameters and Descriptors are required to have a name. 

However, it makes a lot of sense to create a Parameter without having to name it. For example, 

`a=Parameter(2.0,unit='meV')`

seems like a very reasonable thing to write. We would then automatically generate a name for a. For arithmetic operations, the name of the newly generated Parameters or DescriptorNumbers is their unique id. It makes sense to do the same thing here, i.e. the expected result would be
`<Parameter 'Parameter_0': 2.0000 meV, bounds=[-inf:inf]>`

The problem with doing this is that name is the first argument of Parameter and DescriptorNumber. Making name optional means value becomes the first argument, and that breaks a lot. Do we want to do this?

**Comment by damskii9992** [comment]
I do think it is weird to have a mandatory name for all our parameters/descriptors. When I worked on the `unique_name` implementation I attempted to use the regular name as the unique_name because it made sense, unfortunately this caused a host of issues and I ended up adding the extra `unique_name` property instead. I would suggest that:

We remove the "name" parameter completely, I don't see the point of it.
Use the `unique_name` wherever the name is currently used. Possibly rename it?
Add an optional `display_name` or `pretty_name` for names to be presented in the GUI or elsewhere.
Use the `unique_name` in case the `pretty_name` or `display_name` is not set.

These changes WILL break a lot of stuff, so unfortunately this is no small task, despite its seemingly innocently looking simplicity. Since this task is in no way urgent or important, I will therefore claim that it should be postponed to a time when there is time for such less-important stuff.

**Comment by damskii9992** [comment]
Closing this discussion as it will be handled by #161 when we are ready to fully move to the new base classes. 
When Parameters and Descriptors inherit from the new base class they will no longer need a name and the implementation I suggested will be in place.
